src/openzaak/components/documenten/api/serializers.py

```diff
@@ -438,3 +442,55 @@ class Meta:
             "url": {"lookup_field": "uuid"},
             "informatieobject": {"validators": [IsImmutableValidator()]},
         }
+
+
+class ObjectInformatieObjectSerializer(serializers.HyperlinkedModelSerializer):
+    informatieobject = EnkelvoudigInformatieObjectHyperlinkedRelatedField(
+        view_name="enkelvoudiginformatieobject-detail",
+        lookup_field="uuid",
+        queryset=EnkelvoudigInformatieObject.objects,
+        help_text=get_help_text(
+            "documenten.ObjectInformatieObject", "informatieobject"
+        ),
+    )
+    object = LengthHyperlinkedRelatedField(
+        min_length=1,
+        max_length=1000,
+        view_name="",
+        queryset=QuerySet(),
+        lookup_field="uuid",
+        help_text=_(
+            "URL-referentie naar het gerelateerde OBJECT (in deze of een andere API)."
+        ),
+    )
+
+    def set_object_properties(self, object_type):
+        object_field = self.fields["object"]
+        if object_type == "besluit":
+            object_field.view_name = "besluit-detail"
+            object_field.queryset = Besluit.objects
+        else:
+            object_field.view_name = "zaak-detail"
+            object_field.queryset = Zaak.objects
+
+    def to_internal_value(self, data):
+        object_type = data.get("object_type")
+        self.set_object_properties(object_type)
+        res = super().to_internal_value(data)
+        return res
+
+    def to_representation(self, instance):
+        object_type = instance.object_type
+        self.set_object_properties(object_type)
+        return super().to_representation(instance)
+
+    class Meta:
```

Comment:
code style: can you define the meta class _before_ any methods?

Reply:
done